week3/LESSONPLAN.md

```diff
@@ -0,0 +1,44 @@
+# Lesson Plan Week 3
+
+## Agenda
+
+The purpose of this class is to teach the student about more advantage React usage and how to connect a React client to a Express-based Node.js server. To make this more realistic (and efficient) the student will be using `create-react-app`, which will make the workflow easier.
+
+The student will learn about `component lifecycle` and and how to use `componentDidMount()`. The student will make an `API call` using this lifecycle method.
+
+Lastly, the student will learn how to connect a React frontend to a Node.js backend.
+
+## Core Concepts
+
+1. create-react-app
+
+- Generates an fully functional out-of-the-box React environment
+- Solves problem of manually configuring file bundler (like Webpack) or transpiler (Babel)
+
+_Explain to the student the problems Webpack (ex. React necessitates file bundling to work) and Babel (ex. React makes use of ES2017+ features that need to be converted into syntax that older browsers can understand) solves_
+
+- Improves workflow through `hot reloading`
+- Gives a more realistic folder structure
+
+_Show the student how to setup a basic CRA application_
```

Comment:
we wanted to introduce cra in first week itself

Reply:
As mentioned above, the lesson plans are unchanged so we need to rewrite based on trying out the new curriculum.